dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/FdcIntegrationTest.java

```diff
@@ -363,24 +403,37 @@ void givenRequestedFdcContributions_whenProcessDailyFilesFailsToSend_thenTheirSt
 		// Fetch some items of information from the maat-api to use during validation:
 		final var fdcContributions = updatedIds.stream().map(testDataClient::getFdcContribution).toList();
 
-		softly.assertThat(watched.getGlobalUpdateResponse().isSuccessful()).isTrue(); // 1
-		softly.assertThat(updatedIds).hasSize(3).doesNotContainNull(); // 2.
-		softly.assertThat(watched.getRequestedIds()).containsAll(updatedIds); // 3.
-		softly.assertThat(watched.getSentIds()).containsAll(updatedIds); // 4.
-
-		if (watched.getRecordsSent() != 0) { // 5.
-			// contribution_file got created:
-			softly.assertThat(watched.getRecordsSent()).isPositive();
-			softly.assertThat(watched.getXmlCcIds()).doesNotContainAnyElementsOf(updatedIds);
-			softly.assertThat(watched.getXmlFileName()).isNotBlank();
-			softly.assertThat(watched.getXmlFileResult()).isNotNull();
-		} else {
-			softly.assertThat(watched.getXmlCcIds()).isNull();
-			softly.assertThat(watched.getXmlFileName()).isNull();
-			softly.assertThat(watched.getXmlFileResult()).isNull();
+		softly.assertThat(watched.getGlobalUpdateResponse().isSuccessful()).isTrue();
+		softly.assertThat(updatedIds).hasSize(3).doesNotContainNull();
+
+		if (updatedIdsShouldBeRequested)
+			softly.assertThat(watched.getRequestedIds()).containsAll(updatedIds);
+		else
+			softly.assertThat(watched.getRequestedIds()).doesNotContainAnyElementsOf(updatedIds);
+
+		if (updatedIdsShouldBeSent)
+			softly.assertThat(watched.getSentIds()).containsAll(updatedIds);
+		else
+			softly.assertThat(watched.getSentIds()).doesNotContainAnyElementsOf(updatedIds);
+
+		if (checkContributionFile) {
+			if (watched.getRecordsSent() != 0) {
+				// contribution_file got created:
+				softly.assertThat(watched.getRecordsSent()).isPositive();
+				softly.assertThat(watched.getXmlCcIds()).doesNotContainAnyElementsOf(updatedIds);
+				softly.assertThat(watched.getXmlFileName()).isNotBlank();
+				softly.assertThat(watched.getXmlFileResult()).isNotNull();
+			} else {
+				softly.assertThat(watched.getXmlCcIds()).isNull();
+				softly.assertThat(watched.getXmlFileName()).isNull();
+				softly.assertThat(watched.getXmlFileResult()).isNull();
+			}
 		}
-
-		fdcContributions.forEach(fdcContribution -> // 6.
-			softly.assertThat(fdcContribution.getStatus()).isEqualTo(FdcContributionsStatus.REQUESTED));
+		fdcContributions.forEach(fdcContribution -> {
+			softly.assertThat(fdcContribution.getStatus()).isEqualTo(fdcContributionsStatusExpected);
+			if (!checkContributionFile)
+				softly.assertThat(fdcContribution.getContFileId()).isNull();
```

Comment:
This is why I said
> Is there a better name for checkContributionFile since it seems to actually mean don't check that it's null?
Maybe it just needs the reverse condition adding (if that doesn't break the tests)?